ras_backstage/controllers/party_controller.py

```diff
@@ -34,4 +33,17 @@ def get_party_by_respondent_id(party_id):
         raise ApiError(url, response.status_code)
 
     logger.debug('Successfully retrieved respondent party', party_id=party_id)
-    return json.loads(response.text)
+    return response.json()
+
+
+def get_party_by_ru_ref(ru_ref):
+    logger.debug('Retrieving reporting unit', ru_ref=ru_ref)
+    url = f'{app.config["RAS_PARTY_SERVICE"]}party-api/v1/parties/type/B/ref/{ru_ref}'
+    response = request_handler('GET', url, auth=app.config['BASIC_AUTH'])
+
+    if response.status_code != 200:
```

Comment:
We should be using response.raise_for_status and allowing specific error handlers to handle cases where responses aren't in the 200 range.

Reply:
I'm really not a fan of the generic ApiError, it doesn't give us any scope for graceful degradation of service

Reply:
My initial reason to add ApiError in was to make it easier to debug which external call has failed in backstage

For quite a few of the backstage endpoints it makes several external calls and the ApiError handler makes it very clear which service it has failed to call

I'm open to removing it though, it would probably need it's own PR though to do that across the repo as it is pretty widespread